File: LES_EXAMENS/EXAMEN_RATT_2023_2024/exercice_E.c
#include <stdio.h>
#include <string.h>
int max_taille,lettres_conteure,mot_taille[5];
char max_mot[15],mot[5][15],T[10];
int main(){
    for(int i=0;i<10;i++){
        while (1){
            printf("saisir le caractere numero %d minuscule: ",i+1);
            scanf(" %c", &T[i]);
            if(T[i]>='a' && T[i]<='z'){
                break;
            }
        }
    }
    for(int i=0;i<5;i++){
        do{
            printf("saisir le mot %d: ",i+1);
            scanf("%14s", mot[i]);
            mot_taille[i]=strlen(mot[i]);
            lettres_conteure=0;
            for(int l=0;l<mot_taille[i];l++){
                for(int j=0;j<10;j++){
                    if(mot[i][l]==T[j]){
                        lettres_conteure++;
                    }
                }
            }
        }while(!(lettres_conteure==mot_taille[i]));
    }
    max_taille=mot_taille[0];
    strcpy(max_mot, mot[0]);
    for(int i=1;i<5;i++){
        if(mot_taille[i]>max_taille){
            max_taille=mot_taille[i];
            strcpy(max_mot, mot[i]);
        }
    }
    printf("le plus long mot parmi les 5 et son nombre de caractères est %s de taille %d\n",max_mot,max_taille);
    return 0;
}

File: LES_EXAMENS/EXAMEN_RATT_2023_2024/exercice_E.py
mot_taille=[None for i in range(5)]
max_mot=[None for i in range(15)]
mot=[[None for i in range(15)] for j in range(5)] 
T=[None for i in range(10)]
for i in range(0,10):
    while True:
        T[i]=str(input(f"saisir le caractere numero {i+1} minuscule: "))[0]
        if(T[i]>='a' and T[i]<='z'):
            break
for i in range(0,5):
    while True:
        mot[i]=str(input(f"saisir le mot {i+1}: "))
        mot_taille[i]=len(mot[i])
        lettres_conteure=0
        for l in range(0,mot_taille[i]):
            for j in range(0,10):
                if(mot[i][l]==T[j]):
                    lettres_conteure+=1
        if(lettres_conteure==mot_taille[i]):
            break            
max_taille=mot_taille[0]
max_mot=mot[0]
for i in range(1,5):
    if(mot_taille[i]>max_taille):
        max_taille=mot_taille[i]
        max_mot=mot[i]
print(f"le plus long mot parmi les 5 et son nombre de caractères est {max_mot} de taille {max_taille}")